spyder/preferences/languageserver.py

```diff
@@ -608,30 +608,392 @@ def mouseDoubleClickEvent(self, event):
         self.delete_btn = QPushButton(_("Delete currently selected server"))
         self.delete_btn.setEnabled(False)
-        server_group = QGroupBox(_('Available LSP Servers'))
-
-        vlayout = QVBoxLayout()
-        vlayout.addWidget(server_group)
-        # vlayout.addWidget(server_settings_description)
-        vlayout.addWidget(self.table)
-        vlayout.addWidget(self.new_btn)
-        vlayout.addWidget(self.delete_btn)
-        vlayout.addWidget(self.reset_btn)
-        self.setLayout(vlayout)
 
+        # Slots connected to buttons
         self.new_btn.clicked.connect(self.create_new_server)
         self.reset_btn.clicked.connect(self.reset_to_default)
         self.delete_btn.clicked.connect(self.delete_server)
 
+        # Buttons layout
+        btns = [self.new_btn, self.delete_btn, self.reset_btn]
+        buttons_layout = QGridLayout()
+        for i, btn in enumerate(btns):
+            buttons_layout.addWidget(btn, i, 1)
+        buttons_layout.setColumnStretch(0, 1)
+        buttons_layout.setColumnStretch(1, 2)
+        buttons_layout.setColumnStretch(2, 1)
+
+        # Combined layout
+        servers_widget = QWidget()
+        servers_layout = QVBoxLayout()
+        servers_layout.addSpacing(-10)
+        servers_layout.addWidget(servers_label)
+        servers_layout.addWidget(table_group)
+        servers_layout.addSpacing(10)
+        servers_layout.addLayout(buttons_layout)
+        servers_widget.setLayout(servers_layout)
+
+        # --- Tabs organization ---
+        tabs = QTabWidget()
+        tabs.addTab(self.create_tab(introspection_group, advanced_group),
+                    _('Introspection'))
+        tabs.addTab(self.create_tab(linting_widget), _('Linting'))
+        tabs.addTab(self.create_tab(code_style_widget), _('Code style'))
+        tabs.addTab(self.create_tab(docstring_style_widget),
+                    _('Docstring style'))
+        tabs.addTab(self.create_tab(advanced_widget),
+                    _('Advanced'))
+        tabs.addTab(self.create_tab(servers_widget), _('Other languages'))
+
+        vlayout = QVBoxLayout()
+        vlayout.addWidget(tabs)
+        self.setLayout(vlayout)
+
+    @Slot(str)
+    def setup_docstring_style_convention(self, text):
+        """Handle convention changes."""
+        if text == 'Custom':
+            # Show select and ignore options
+            self.docstring_style_select.label.setVisible(True)
+            self.docstring_style_select.textbox.setVisible(True)
+            self.docstring_style_ignore.label.setVisible(True)
+            self.docstring_style_ignore.textbox.setVisible(True)
+
+            # Hide add_select and add_ignore options
+            self.docstring_style_add_select.label.setVisible(False)
+            self.docstring_style_add_select.textbox.setVisible(False)
+            self.docstring_style_add_ignore.label.setVisible(False)
+            self.docstring_style_add_ignore.textbox.setVisible(False)
```

Comment:
@dalthviz, what we agreed to do with @CAM-Gerlach is the following:

1. Remove `docstring_style_add_select` and `docstring_style_add_ignore` options, and only leave `docstring_style_select` and `docstring_style_ignore`.
2. When `Numpy` or `PEP 257` are selected as conventions, then:
    1. The labels on those options must display `Show the following errors in addition to the specified convention:` and `Ignore the following errors in addition to the specified convention:`.
    2. The values saved on those options need to be passed to `addSelect`/`addIgnore` in the `update_python_config` method of `LSPManager`.
    3, `select` and `ignore` in `update_python_config` need to be made empty.
3. When `Custom` is selected, then:
    1. The labels must display `Show the following errors:` and `Ignore the following errors:`.
    2. The values of those options need to be passed to `select` and `ignore` in  `update_python_config`.
    3. `addSelect`/`addIgnore` in `update_python_config` need to be made empty.

I know this is a bit confusing to explain. Please let me know if it's clear for you or you need me to clarify things a bit more.